src/hooks/screen/useCreatePostHook.js

```diff
@@ -11,11 +11,15 @@ import AnonUserInfoRepo from '../../service/repo/anonUserInfoRepo';
  */
 const useCreatePostHook = (isAnonymous) => {
   const {params = {}} = useRoute();
-  const {topic} = params;
+  const {topic, isCreateCommunity} = params;
   const [selectedTopic, setSelectedTopic] = React.useState(topic);
   const [anonUserInfo, setAnonUserInfo] = React.useState(null);
   const [headerTitle, setHeaderTitle] = React.useState(
-    selectedTopic ? `Create Post in #${selectedTopic}` : 'Create Post'
+    isCreateCommunity
+      ? 'Create first post'
+      : selectedTopic
+      ? `Create Post in #${selectedTopic}`
+      : 'Create Post'
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for setting the `headerTitle` state seems to be a bit complex and could be simplified. Also, it's not clear what happens when both `isCreateCommunity` and `selectedTopic` are true. It would be better to separate this into a function that clearly defines the precedence of these conditions.
```diff
-   const [headerTitle, setHeaderTitle] = React.useState(
-     isCreateCommunity
-       ? 'Create first post'
-       : selectedTopic
-       ? `Create Post in #${selectedTopic}`
-       : 'Create Post'
-   );
+   const getInitialHeaderTitle = () => {
+     if (isCreateCommunity) return 'Create first post';
+     if (selectedTopic) return `Create Post in #${selectedTopic}`;
+     return 'Create Post';
+   };
+
+   const [headerTitle, setHeaderTitle] = React.useState(getInitialHeaderTitle());
```
This way, we have a clear order of precedence: `isCreateCommunity` takes priority over `selectedTopic`, and if neither is true, we default to 'Create Post'.


<!-- This is an auto-generated comment by OSS CodeRabbit -->